vaadin-dialog-flow-parent/vaadin-dialog-flow/src/main/java/com/vaadin/flow/component/dialog/Dialog.java

```diff
@@ -620,7 +620,7 @@ private DialogFooter(Dialog dialog) {
      * components will be attached to as well as the renderer function used by
      * the dialog.
      */
-    abstract static class DialogHeaderFooter implements Serializable {
+    abstract static class DialogHeaderFooter implements HasComponents {
```

Comment:
Looks like several more APIs from `HasComponents` (add with text, add with collection, same with remove, ...) need to be overriden to then either initialize or remove the client-side renderer.

Reply:
Added the missing `HasComponents` APIs.